samcli/lib/delete/cf_utils.py

```diff
@@ -0,0 +1,102 @@
+"""
+Delete Cloudformation stacks and s3 files
+"""
+
+import logging
+
+from botocore.exceptions import ClientError, BotoCoreError
+from samcli.commands.delete.exceptions import DeleteFailedError
+
+LOG = logging.getLogger(__name__)
+
+
+class CfUtils:
+    def __init__(self, cloudformation_client):
+        self._client = cloudformation_client
+
+    def has_stack(self, stack_name):
+        """
+        Checks if a CloudFormation stack with given name exists
+
+        :param stack_name: Name or ID of the stack
+        :return: True if stack exists. False otherwise
+        """
+        try:
+            resp = self._client.describe_stacks(StackName=stack_name)
+            if not resp["Stacks"]:
+                return False
+
+            stack = resp["Stacks"][0]
+            return stack["StackStatus"] != "REVIEW_IN_PROGRESS"
```

Comment:
Why is this a special case?

Reply:
My understanding was that when a stack is in review on CF, the change_set is not executed yet and so if a user wants to delete this stack, we will have to handle this separately(deleting change_set and not stack). Just to avoid that for now, I check the status. Correct me if my understanding is wrong.

Reply:
Not sure. I'd assume any stack that exists regardless of status should be deletable. In any case, whenever the _why_ isn't clear in the code, please add a comment for explanation.